src/System.Private.CoreLib/shared/System/Threading/Timer.cs

```diff
@@ -477,7 +477,7 @@ internal bool Change(uint dueTime, uint period)
                         // Don't emit this event during EventPipeController.  This avoids initializing FrameworkEventSource during start-up which is expensive relative to the rest of start-up.
                         !EventPipeController.Initializing &&
 #endif
-                        FrameworkEventSource.IsInitialized && FrameworkEventSource.Log.IsEnabled(EventLevel.Informational, FrameworkEventSource.Keywords.ThreadTransfer))
+                        FrameworkEventSource.Log.IsEnabled(EventLevel.Informational, FrameworkEventSource.Keywords.ThreadTransfer))
```

Comment:
Why is it ok to delete the `FrameworkEventSource.IsInitialized` check?

Reply:
`IsInitialized` check was checking if static ctor of this type run successfully, considering this type is created from another static ThreadPoolGlobals ctor I don't think we can recover from such condition anyway

Reply:
ok

Reply:
`EventSource` is a heavy type and the pervasive use of it in common code paths is one of the reasons why a CoreRT Hello World has 4 megabytes (EventSource brings the entire reflection stack into the image). I imaging it will be of similar concern to Mono. I've been looking into ways to get this linked out in the past.

Deleting the place where the FrameworkEventSource is allocated using a [removable feature](https://github.com/dotnet/designs/pull/42) annotation would likely get us close to removing all of it. The `FrameworkEventSource.IsInitialized` checks would help existing code not to AV when that happens.

I guess we can restore this when needed, but not deleting in the first place would save some work.

Reply:
IL linker can remove EventSource like types if used with expected code pattern and you enable the removal, we already used it for example for wasm. I didn't check this specific code pattern and we'll do such things later in the process but it should not be hard to expand the detection anyway.

Reply:
> if used with expected code pattern

Is there a general purpose mechanism for that or would this require hardcoding knowledge of FrameworkEventSource into IL Linker?

Reply:
IL Linker knows only about EventSource type, any type deriving from it should be covered automatically

Reply:
Sounds good!